bst_plugin/process_nst_compute_group_ttest.m

```diff
@@ -0,0 +1,251 @@
+    
+    % === TAIL FOR THE TEST STATISTIC
+    sProcess.options.tail.Comment  = {'One-tailed (-)', 'Two-tailed', 'One-tailed (+)', ''; ...
+                                      'one-', 'two', 'one+', ''};
+    sProcess.options.tail.Type     = 'radio_linelabel';
+    sProcess.options.tail.Value    = 'two';
+    
+    
+end
+
+
+%% ===== FORMAT COMMENT =====
+function Comment = FormatComment(sProcess) 
+    Comment = sProcess.Comment; 
+end
+
+function OutputFiles = Run(sProcess, sInputs)
+    OutputFiles={};
+    
+    % parse input : 
+
+	for i=1:numel(sInputs)
+        s_B(i)=in_bst_data(sInputs(i).FileName);
+        B(:,i)=s_B(i).Value; % might need to transpose. Have to check how cB is registered 
+	end     
+
+    n_chan=size(B,1);
+    n_subject=size(B,2);
+    
+    switch sProcess.options.method.Value 
+        case 'mean' 
+            [t,df]=compute_mean(B);
+    end    
+
+    % Calculate p-values from t-values
+    p = ComputePvalues(t, df, 't',   sProcess.options.tail.Value );
+   
+    % Change folder to inter study. 
+    iStudy = sInputs(1).iStudy;
+
+    [tmp, iSubject] = bst_get('Subject', sInputs.SubjectName);
+    [sStudyIntra, iStudyIntra] = bst_get('AnalysisInterStudy', iSubject);
+    
+    [ChannelFile] = bst_get('ChannelFileForStudy', iStudy);
+    [tmp, iChannelStudy] = bst_get('ChannelForStudy', iStudyIntra);
+    db_set_channel(iChannelStudy, ChannelFile, 0, 0);
+    
+    
+    % === OUTPUT STRUCTURE ===
+    % Initialize output structure
+
+    sOutput = db_template('statmat');
+    sOutput.pmap         = [p;p]';
+    sOutput.tmap         = [t;t]';
+    sOutput.df           = ones(n_chan,1)*df;
+    sOutput.ChannelFlag= ones(1,n_chan);
+    sOutput.Correction   = 'no';
+    sOutput.Type         = 'data';
+    sOutput.Time         = [1];
+    sOutput.ColormapType = 'stat2';
+    sOutput.DisplayUnits = 't';
+    sOutput.Options.SensorTypes = 'NIRS';
+
+    
+    % Formating a readable comment such as -Rest +Task
+    comment=['Group Stat (' num2str(n_subject) ' subjects), '] ;
+    comment=[comment  s_B(1).Comment(1:end-2) ];
+
+    switch sProcess.options.tail.Value 
+        case {'one-'}
+             comment=[ comment ' < 0 '];      
+        case {'two'}
+             comment=[ comment ' <> 0 ']; 
+        case { 'one+'}
+             comment=[ comment ' > 0 ']; 
+    end  
+    
+    sOutput.Comment=comment;
+    sOutput = bst_history('add', sOutput, s_B(1).History, '');
+
+    sOutput = bst_history('add', sOutput, 'Group stat', comment);
+    OutputFiles{1} = bst_process('GetNewFilename', fileparts(sStudyIntra.FileName), 'pdata_ttest_matrix');
+    save(OutputFiles{1}, '-struct', 'sOutput');
+    db_add_data(iStudyIntra, OutputFiles{1}, sOutput);
+
+end
+
+function [t,df]=compute_mean(B)
+  
+    n_chan=size(B,1);
+    n_subject=size(B,2);
+
+    mean_B= mean(B,2);
+    varB=  var(B,0,2) ;
+
+    df=n_subject-1;
+    t=zeros(1,n_chan);
+    
+    for i = 1:n_chan
+        t(i)= mean_B(i) / sqrt(varB(i)) ; 
```

Comment:
maybe it's safer to use a more general formula including the variance of X. The current is OK if you're doing only a mean but it won't if you include other group-level regressors like performance.